Data contribution 2 (#37)
arcus/Arcus_Labs_Orientation: data_contribution_2_data.md

<!--

author:   Arcus Library Sciences
email: dlarcuslibraryscience@chop.edu
version:  1.0.0
current_version_description: Brief description of why this version exists
module_type: standard
docs_version: 1.1.0
language: en
narrator: UK English Female
title: Arcus Project Template Orientation

> comment:  Arcus Archives serves as the canonical repository for research data at CHOP. Archiving data with Arcus safeguards and maintains research data in accordance with established archival protocols, also while promoting data sharing. The Data Contribution Module provides an overview of the reasons to archive research data with Arcus, as well as the processes involved in scoping data, conducting privacy assessments, and addressing technical requirements prior to data submission. This Module explores the Project Template directory structure, and delves into the specific steps required for archiving data within the Arcus platform.

Reviewer: ```suggestion
comment: Learn about how to disclose data to CHOP by making a data contribution!
long_description:  Arcus Archives serves as the canonical repository for research data at CHOP. Archiving data with Arcus safeguards and maintains research data in accordance with established archival protocols, also while promoting data sharing. The Data Contribution Module provides an overview of the reasons to archive research data with Arcus, as well as the processes involved in scoping data, conducting privacy assessments, and addressing technical requirements prior to data submission. This Module explores the Project Template directory structure, and delves into the specific steps required for archiving data within the Arcus platform.
```

Note that you're missing a "long_description" so I've taken the liberty of adding a comment and making your comment the long_description.  Feel free to tweak, this is just my suggestion.

### Future data contributors

This module is an overview of CHOP's Project Template structure, all contributed data is arranged in this structure for archiving in Arcus. The Project Template is useful at all stages of research, we suggest implementing it as early as possible in your research as it provides a shared, documented framework for organizing a research effort.

Arcus's Library Science team is happy to meet with researchers at all phases of research for research data management consultations and planning for future archival contributions. This includes early in your research project, as we can help set up a project template file directory structure for storing data and recommend metadata and organization best practices. In addition to this module, there are additional data management resources available on CHOP’s [Arcus resources page](https://www.research.chop.edu/applications/arcus/resources).

> If after viewing this module, you are prepared to archive data with Arcus, please fill out [the following request](https://pm.arcus.chop.edu/servicedesk/customer/portal/6/create/256) to start the process.

Reviewer: ```suggestion
If after viewing this module, you are prepared to archive data with Arcus, please fill out [a data contribution request](https://pm.arcus.chop.edu/servicedesk/customer/portal/6/create/256) to start the process.
```
When I saw the word "following" I was looking for another line...

As part of the Arcus Lab deployment, your team should have received a Project Template orientation by a member of the Library Science team. If you missed the orientation, or need a refresher on the orientation, please see this [video of the orientation](https://www.youtube.com/watch?v=YJvA_cryI1s). Much of the content covered in the video is also in this module.

When appropriate, archiving your research in Arcus is expected from Scientific Projects with an Arcus Lab. This is documented in the [Arcus Terms of Use](https://arcus.chop.edu/terms-of-use). Archiving is required if you would like to move any data created within an Arcus Lab to a new Scientific Project or if other research teams would like to reuse your data.

When you are ready to archive your lab data, please submit the following request in the [Arcus Help Center](https://pm.arcus.chop.edu/servicedesk/customer/portal/6/create/256) to begin the data contribution process.

## Arcus Data Lifecycle

> Arcus' goal for research data management and the project template is to provide tools that are relevant throughout the entire lifecycle of research data. The project template is designed to be adaptable and iterative to capture the wide range of research activities at CHOP. The project template combines flexibility to encompass diverse data capturing needs, while also maintaining a consist structure for all archived data which facilitates seamless communication among various projects spanning different domains, thereby promoting effective data sharing.

Reviewer: ```suggestion
Arcus's goal for research data management and the project template is to provide tools that are relevant throughout the entire lifecycle of research data. The project template is designed to be adaptable and iterative to capture the wide range of research activities at CHOP. The project template combines flexibility to encompass diverse data capturing needs, while also maintaining a consist structure for all archived data which facilitates seamless communication among various projects spanning different domains, thereby promoting effective data sharing.
```
I *think* the current "possessive apostrophe following S" rule includes another s?  But feel free to disregard, I haven't actually researched this.

**How was this structure developed?**

The CHOP project template file directory structure was adapted from [DrivenData’s Cookiecutter Data Science template](https://cookiecutter.readthedocs.io/en/stable/). It was adapted by former Arcus Digital Archivist, Christiana Dobrzynski, and former CHOP Bioinformaticist, Perry Evans. Both Arcus’s and DrivenData’s templates aim to organize research data and tools for accuracy and reproducibility. See [DataDriven's introduction](http://drivendata.github.io/cookiecutter-data-science/) to learn more about the goals and purpose of project template structures for data preservation and sharing.

The CHOP project template evolved through iterations and feedback from CHOP researchers. A multi-disciplinary groups of practicioners were consulted in the template adaptation development, including:

- Bioinformatics
- Cancer research
- Microbiome center
- Research IT
- Clinical sequening unit
- Medical Informatics Unit

**How is the Project Template used at Arcus?**

> The Project Template prioritizes streamlined archiving and reproducible research pathways. It archives a wide range of research types from the Research Institute, making them discoverable through tools like [Arcus Cohort Discovery](https://arcus.chop.edu), [Gene](https://chop.alationcloud.com), and [Omics Variant Browser](https://arcus.chop.edu). The Project Template facilitates organizing diverse research data in a single directory structure, enabling automated archiving, metadata management, and data delivery throughout the research data lifecycle. This file directory structure is used for the entire lifecycle of research data within Arcus:

Reviewer: These links don't actually take you into the applications, so consider something like:

```suggestion
The Project Template prioritizes streamlined archiving and reproducible research pathways. It archives a wide range of research types from the Research Institute, making them discoverable through tools like Arcus Cohort Discovery, available as an application [on the Arcus website](https://arcus.chop.edu), [Gene](https://chop.alationcloud.com), and Omics Variant Browser, [available as an application [on the Arcus website](https://arcus.chop.edu). The Project Template facilitates organizing diverse research data in a single directory structure, enabling automated archiving, metadata management, and data delivery throughout the research data lifecycle. This file directory structure is used for the entire lifecycle of research data within Arcus:
```

> ![Graphic representation of the lifecycle of the project template in Arcus](media/project_template/ProjectTemplate_Lifecycle.png)

Reviewer: Ideally any text on this image (and other images that similarly convey information via text) is captured in alt text or other text elsewhere in the document.  This is important for accessibility, mobile viewing where an image might be v small on a phone, as well as search.  But don't let this hold you back, we can't let the perfect be the enemy of the good!

Reviewer 2: Thank you for pointing this out Joy! We reused graphics we already has for this module, which is why there are so many images with text. We are not going to be able to lay these images out in CSS to match W3 standards. I went through and at least improved all the alt text. We will keep in mind these alt text standards in future.

The project template provides a shared structure so that institutional knowledge previously held locally by various members of the data creation team becomes centralized.

The utility of [the project template](https://github.research.chop.edu/arcus/rdm-project-template) for lab drive organization and integration with the Arcus archives is summarized in the graphic below.

> ![Graphic representation of the workflow of research data and the project tempalte](media/project_template/ProjectTemplate_workflow.png)

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

## Project Template

The project template structure includes directories for capturing three major aspects of a research effort: the data (data), the tools needed to work with that data (access tools), and the contextual information needed to understand the effort and its constituent parts (contextual). The high level directories are as follows (items with asterix are required):

- **_Configs_** (contextual)
- **_Data_** (data)\*
- **_Manifests_** (data)\*
- **_Models_** (access tools)
- **_References_** (contextual)
- **_Reports_** (contextual)
- **_Requirements_** (contextual)
- **_SRC_** (access tools)

Below is an image of the entire Project Template Directory, with more detail about each section:

> ![Graphic respresentation of the Project template with a short explanation of all the different sections](media/project_template/ProjectTemplate_Description.png)

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases.  I leave it to you to decide if the effort of re-doing is worth it.

### Research Data

The Project Template brings together three categories of information: Research Data, Access Tools and Contextual files. Research data is the actual data collected during the course of research processes used for analysis. The manifests describe this data, crosswalking files to participants. Research data (with manifests) is the mininum required information for all Arcus data contributions.

> ![Graphic respresentation of the Project template with that denotes research data sections](media/project_template/research_data.png)

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

### Access Tools

Access Tools are the code used to do the analysis. This can include machine models, scripts and Jupyter notebooks.

> ![Graphic respresentation of the Project template with that denotes access tool sections](media/project_template/access_tools.png)

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

### Contextual Files

Contextual Files provide information needed to understand the data and analysis. This can include omics protocols, data dictionaries, reports and diagrams.

> ![Graphic respresentation of the Project template with that denotes contextual files sections](media/project_template/contextual_files.png)

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

All files within the **data/** folder and its subdirectories will be listed in the **file_manifest.csv**. The manifests are detailed in the [manifests](#manifests) section of this course.

### data/raw

This directory holds authoritative source data that should never be deleted. This folder is where the original, unmodified data for the research project is stored. In a research process, this is the data used for the initial analysis. Further sub-directories can be added to organize data, if necessary.

**Within an [Arcus Scientific Lab](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/Arcus_Labs_Orientation/main/arcus_orientation.md#1)**

- Arcus delivered archival data will be found here.
- Study team generated data brought into Arcus goes here.
- This data is managed by Arcus, and should not be modified by the research team.

**Raw** data is different depending on the type of research. Please select what type(s) of data you would like more information about, you can select both:

- [ ] omics data
- [ ] clinical data

> <script output="data_type">"@input"</script>

Reviewer: Clicking these, I'm left with a big empty space afterwards that looks like nothing happened unless I scroll down the page.  Consider checking the functions?  I think there may be a missing image or one that is being hidden?

<script modify="false">
try {
  let data_type = @input(`data_type`)

  if(data_type[0]) {
    send.liascript(`## Omics Data

> ![Omics data/raw](media/project_template/raw_omics.gif)

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

For omics data, the raw folder contains the sequencing data. Most sequencing providers will generate a [**fastq**](https://maq.sourceforge.net/fastq.shtml) file or [**cram**](https://samtools.github.io/hts-specs/CRAMv3.pdf) file, we prefer these filetypes for archiving.
* These files contain genomic sequences called reads. With paired reads, there are two fastq files per sample. Cram files are single files aligned to a reference genome.

- We collect the compressed fastq form fastq.gz, which can be made with lossless compression utility tools like gzip. fastq metadata are described in the fastq directory.
- Cram files are human readable and highly space efficient by using reference-based compression of sequence data. These files enable us to run a complete re-analysis of the data. Cram files require a companion index file, .crai.

`)
  } else send.clear()
} catch(e) { }
</script>

<script modify="false">
try {
  let data_type = @input(`data_type`)

  if(data_type[1]) {
    send.liascript(`## Clinical Data

> ![Clinical data/raw](media/project_template/raw_clinical.gif)

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

    send.liascript(`**Nothing is selected** 
<br>

`)
  } else send.clear()
} catch(e) { }
</script>

### data/interim

The interim directory is for storing outputs of data processing and analysis completed using the original, unmodified data store in **data/raw**. It is generally used for files that do not need to be stored long-term. Further sub-directories can be added to organize data, if necessary.

**Within an [Arcus Scientific Lab](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/Arcus_Labs_Orientation/main/arcus_orientation.md#1)**

- Data in this directory is managed by the study team
- It should be used as an unregulated space for intermediate and temporary files.
- Recommend establishing retention schedules for regular review/clean-up of data in this folder.

**Interim** data is different depending on the type of research. Please select what type(s) of data you would like more information about, you can select both:

- [ ] omics data
- [ ] clinical data
<script output="data_type_interim">"@input"</script>

<script modify="false">
try {
  let data_type = @input(`data_type_interim`)

  if(data_type[0]) {
    send.liascript(`
    ## Omics Data

> ![Omics data/interim](media/project_template/interim_omics.gif)

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

This directory is for the quality control and other reporting created during a bioinformatics workflow, using the sequence files stored in the \_data/raw\_ directory.

- QC metrics Quality Control (QC) metrics are reported at various stages of analysis pipelines and give information about the quality of the data generated. QC metrics files should be in a tabular file format, with .type\_metrics or .duplicate\_metrics as the extension.
    `)
  } else send.clear()
} catch(e) { }
</script>

<script modify="false">
try {
  let data_type = @input(`data_type_interim`)

  if(data_type[1]) {
    send.liascript(`
    ## Clinical Data

> ![Clinical data/interim](media/project_template/interim_clinical.gif)

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

    `)
  } else send.clear()
} catch(e) { }
</script>

### data/endpoints

The endpoints directory holds the final results created as part of a research analysis. Often, these are files created to support papers or grants, and other dissemination. Further sub-directories can be added to organize data, if necessary.

**Within an [Arcus Scientific Lab](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/Arcus_Labs_Orientation/main/arcus_orientation.md#1)**

- Data in this directory is managed by the study team.
- Data in this directory will be saved if the project is archived in Arcus.

**Endpoints** data is different depending on the type of research. Please select what type(s) of data you would like more information about, you can select both:

- [ ] omics data
- [ ] clinical data
<script output="data_type_endpoints">"@input"</script>

<script modify="false">
try {
  let data_type = @input(`data_type_endpoints`)

  if(data_type[0]) {
    send.liascript(`
    ## Omics Data

> ![Omics data/endpoints](media/project_template/endpoints_omics.gif)<!-- style = "max-height: 500px;" -->

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

- **vcf** files usually contain multiple samples, and are the starting point for most reserach project's analysis. They are not appropriate for constructing cohorts from multiple projects because they are missing necessary information contained in the gvcf files. We collect the compressed version of a vcf file (vcf.gz), which can be made with a lossless compression utility tool like gzip[gzip](https://www.gzip.org). The variant calls in these files might differ from those in a gvcf file because they are made by considering information from all samples. A researcher would want to use these files when considering each project separately. All vcf files should include an index file, .tbi

    `)
  } else send.clear()
} catch(e) { }
</script>

<script modify="false">
try {
  let data_type = @input(`data_type_endpoints`)

  if(data_type[1]) {
    send.liascript(`
    ## Clinical Data

> ![Clinical data/endpoints](media/project_template/endpoints_clinical.gif)

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

- The Arcus Applied Data Sciences team offers [GIS Data](https://chop.alationcloud.com/domain/30/) of many high value publicly available regularly refreshed Geographical Information System Datasets.
- There is also an option via [Jira Service Desk](https://jira.arcus.chop.edu:8443/servicedesk/customer/portal/6/create/355?q=upload&q_time=1696276238126) to upload reference data directly to your lab following disclosure and approval.

    `)
  } else send.clear()
} catch(e) { }
</script>

## manifests/

Manifests are an inventory of all data in the collection, and provide a mapping between research data in the data folders, and participant information. The manifests also create a mapping between data and associated pipeline and technical information about workflows. There are three main manifests that are mandatory for every archival collection:

- file_manifest.csv
- participant_manifest.csv
- partcipant-crosswalk.txt

Additional manifests are only required if needed for the data or collection type. These files are detailed in the next sections. The graphic below illustrates the linking between the files:

> ![ID Crosswalks between manifests](media/project_template/manifests_linkages.png)

Reviewer: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

**Within an [Arcus Scientific Lab](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/Arcus_Labs_Orientation/main/arcus_orientation.md#1)**

- Managed by Arcus, you will not need to create these for yourselves
- This will only appear in the lab if archival data is delivered

### manifests/file_manifest

> The file_manifest.csv matches the biosample_id to each file in the data folders. Below is more detail about each section in the file:

Reviewer: ```suggestion
The file\_manifest.csv matches the biosample\_id to each file in the data folders. Below is more detail about each section in the file:
```
If you don't escape underscores, markdown thinks you want the enclosed text italicized if there are two underscores on the same line.

> - biosample_id is an ID number for each file. For some studies, each file is derived about specific biosamples, so we suggest using the sample id. ideally, biosample_id links to the CHOP biobank. When you cannot link the the biobank, treat biosample_id as the IDs you use for samples taken from participants. For studies where there are no biosamples, the biosample_id can be the file name.

Reviewer: ```suggestion
- biosample\_id is an ID number for each file. For some studies, each file is derived about specific biosamples, so we suggest using the sample id. ideally, biosample\_id links to the CHOP biobank. When you cannot link the the biobank, treat biosample\_id as the IDs you use for samples taken from participants. For studies where there are no biosamples, the biosample\_id can be the file name.
```

- file_type is the type of file, indicated by the file extension
- protocol is only for omics data, select the omics data example below for more information

> - file_path is the file path for each file in the **_data\_** folders. File paths should start with **_data\_** and end with the full file name with extension

Reviewer: ```suggestion
- file\_path is the file path for each file in the **\_data\_** folders. File paths should start with **\_data\_** and end with the full file name with extension
```

The **file_manifest.csv** may look different depending on the type of research. Please select below if you need more information about either omics or clinical data for this directory: :

- [ ] omics data
- [ ] clinical data
<script output="data_type_fm">"@input"</script>

<script modify="false">
try {
  let data_type = @input(`data_type_fm`)

  if(data_type[0]) {
    send.liascript(`## Omics Data

> For Omics data, the **file_manifest.csv** matches biosample IDs to data files and experimental protocols, described in _yaml_ files. More information about the protocols files is available in the references section of this module. Many files might share the same experimental protocol. These yaml protocol files describe experiment and data processing details. See below for the columns in a file_manifest:

Reviewer: ```suggestion
For Omics data, the **file\_manifest.csv** matches biosample IDs to data files and experimental protocols, described in \_yaml\_ files. More information about the protocols files is available in the references section of this module. Many files might share the same experimental protocol. These yaml protocol files describe experiment and data processing details. See below for the columns in a file\_manifest:
```

| protocol                | Each experiment template has protocol yaml files or capture kit information used to describe experiment metadata. This column points to the file path of the protocol or the capture kit information for this file. Paths should start with references/procotols/ or data/ref-data/platform-data                                 | String |
| file_path               | Use one file path per row. It should start with data/.                                                                                                                                                                                                                                                                           | String |
| file_groups             | Files in the same group are related. Paired fastq files belong in the same group. A bam file and its index belong in the same group. Plink bfiles belong in the same group.                                                                                                                                                      | String |

> | derived\_from\_file\_group | This column describes relations between file groups. We want to capture consecutive pipeline steps. For example, a bam file is derived from a paired fastq group. Use the name of the file_groups used to construct this file. Delimit multiple groups with a semicolon. Use NA when there are no prior step files to reference. | String |

Reviewer: ```suggestion
| derived\_from\_file\_group | This column describes relations between file groups. We want to capture consecutive pipeline steps. For example, a bam file is derived from a paired fastq group. Use the name of the file_groups used to construct this file. Delimit multiple groups with a semicolon. Use NA when there are no prior step files to reference. | String |
```
Strangely this is still showing up with weird italicizing for me, even though you're escaping.  Not sure what's up?

| biosample_id            | This ID links to PARTICIPANT_MANIFEST.                                                                                                                                                                                                                                                                                           | String |
| file_type               | Each experiment template has a list of required file types. Use those terms.   | String |
| file_path               | Use one file path per row. It should start with data/.                         | String |

    `)
  } else send.clear()
} catch(e) { }
</script>

### manifests/participant_manifest

The participant_manifest.csv identifies which participants information links to each of the files in the file_manifest. Below is more detail about each section of the file:

> - local_participant_id is a local identifier the study team used to identify the patient

Reviewer: ```suggestion
- local\_participant\_id is a local identifier the study team used to identify the patient
```

- The biosample_id will be the same as the one listed in the file_manifests.csv. Linking a local_participant_id to a biosample_id identifies which patients information is related to the file.
- cohort is optional, please fill this in if there is additional cohort information or identification needed.

The **participant_manifest.csv** may look different depending on the type of research. Please select which type of data from below you need more information about for this directory:

- [ ] omics data
- [ ] clinical data
<script output="data_type_pm">"@input"</script>

<script modify="false">
try {
  let data_type = @input(`data_type_pm`)

  if(data_type[0]) {
    send.liascript(`## Omics Data

> The **participant_manifest.csv** matches participants (or patients) to cohorts and biosample IDs from the **file_manifest.csv**. Ideally, biosample\_id links to the CHOP biobank. When you cannot link to the biobank, treat biosample\_id as the IDs you use for samples taken from participants. If you deal with only one sample type, you might use the participant\_id. If you run a treatment/control experiment, you might use participant\_id and treat participant\_id\_control as as a biosample ID scheme. If you work with different tissue samples from participants, you might use participant\_id\_tissue as a biosample ID scheme.

Reviewer: This is still showing up weirdly italicized for me?  Again, not sure what's up

| column               | definition                                                                                                                    | type   |
| -------------------- | ----------------------------------------------------------------------------------------------------------------------------- | ------ |

> | local\_participant\_id | This ID uniquely defined a person, and can be linked to an MRN.                                                               | String |

Reviewer: This is still showing up weirdly italicized for me? Again, not sure what's up

| cohort               | Use this column to group participants into cohorts that will be cwopared (For example, case vs healthy control).              | String |
| biosample_id         | Ideally, this ID can link to the CHOP biobank. When this is not possible, use the sample ID from your project.                | String |
| family_id            | When participants are related, use family_id to group related participants. With trios or duos, the proband ID is often used. | String |

> | family_role          | Use a term from [eHB\_relationship\_types\_as\_of\_10\_30.json](https://github.research.chop.edu/arcus/rdm-project-template/blob/master/manifests/data_dicts/eHB_relationship_types_as_of_10_30.json) to indicate mother, father, proband, sister, etc..                    | String |

Reviewer: This is still showing up weirdly italicized for me? Again, not sure what's up

| -------------------- | --------------------------------------------------------------------------------------------------------------------- | ------ |
| local\_participant\_id | The local id of a participant.                                                                                        | String |
| local\_relative\_id    | The local id of a relative to the participant.                                                                        | String |
| relative\_family\_role | The familial relationship of the relative to the participant. Use terms from eHB\_relationship\_types\_as_of\_10\_30.json. | String |

### manifests/familyid_crosswalk

- This manifest should be used with trio and cohort omics data. A trio will contain three participants, a cohort can contain hundreds. This file walks the name for the trio or cohort file with the local_participant_id's included in it.
- This is also called a PED or pedigree file in bioinformatics workflows. A pedigree is a structured description of the familial relationships between samples, see this [link for more information](https://gatk.broadinstitute.org/hc/en-us/articles/360035531972-PED-Pedigree-format). 

> 

Reviewer: ```suggestion

<!-- data-type="none" -->
```
Do you want the graph type "parallel crosswalk" to appear?  If not, I suggest you add the data type to the top of the table.

Reviewer 2: We did not want a 'parallel crosswalk' and did not know how to get rid of it!

| family_id            | Required, the family_id for the trio data                                           |
| -------------------- | ----------------------------------------------------------------------------------- |
| local\_participant\_id | Required, the local\_particpant\_id for each of the participants included in the trio |
| paternal_id          | Optional, the local\_participant\_id for the father of the participant                |
| maternal_id          | Optional, the local\_participant\_id for the mother of the participant                |
| sex                  | Optional, the sex of the participant. 1 for male, 2 for female                      |

### manifests/file_derivation.csv

The **file_derivation.csv** manifest is only required for omics contributions with multiple filetypes generated through a bioinformatics pipeline or workflow.

**file_derivation.csv** describes the relationships between files in a pipeline or workflow.

| column                 | definition                                                      | type   |
| ---------------------- | --------------------------------------------------------------- | ------ |

> | destination\_file\_group | The files in this file group is derived from source_file_group. | String |

Reviewer: ```suggestion
| destination\_file\_group | The files in this file group is derived from source\_file\_group. | String |
```

> | source\_fil\e_group      | File group used to derive the destination_file_group.           | String |

Reviewer: ```suggestion
| source\_file\_group      | File group used to derive the destination\_file\_group.           | String |
```

### manifests/env manifest

> For each script/notebook in _src/_, and each model in _models/_, there should be an env file (here env refers to a file named env with any extension, so env.yaml or env.txt, for example) that describes the environment in which it was created or run. Environment files should be named as follows: **descriptiveName_env** and placed in a folder called environments within the _configs/environment_ directory. Either individuals files or entire folders (whichever is the appropriate level) in scripts and notebooks within the _src/_ directory, or the _models/_ directory will need to be added to the env manifest file, matching them with their related environment file. See the see below for more infromation about this file:

Reviewer: ```suggestion
For each script/notebook in \_src/\_, and each model in \_models/\_, there should be an env file (here env refers to a file named env with any extension, so env.yaml or env.txt, for example) that describes the environment in which it was created or run. Environment files should be named as follows: **descriptiveName\_env** and placed in a folder called environments within the \_configs/environment\_ directory. Either individuals files or entire folders (whichever is the appropriate level) in scripts and notebooks within the \_src/\_ directory, or the \_models/\_ directory will need to be added to the env manifest file, matching them with their related environment file. See the see below for more information about this file:
``` 

Again, I think I'm interpreting what you're doing... that you want these underscores to appear, right?  If you have two or more in a single line of text, you **must** prefix them with the backslash to escape their normal markdown usage (italics)

Reviewer 2: Thank you for catching these. Nicole and I thought we had corrected all of them, but more keep on showing up!

| --------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------ |
| programming_filegroup | Enter the highest level folder that the environment file relates to. If the file relates to an entire directory then put the whole directory file path. If the file relates to a suddirectory enter that filepath. If it relates to a single file enter the file path and filename. | String |
| related_environment   | Enter the environment filename. Some environment files will be entered multiple times as they relate to multiple files.                                                                                                                                                             | String |
|                       |

## src/

> The src or sources folder stores the access tools required to work with the research data and repeat the analysis. The need for access tools is dependent on the type of research, not all research has rules, scripts or notebooks. Any scripts saved in the src folder require an environment manifest to document the computing environment the code is run in, see the [environment manifest page](#manifestsenv_manifestcsv) for more information. Subdirectory folders can be customized and added as needed, below are the common directories or data types used in scientific research:

Reviewer: Link doesn't go where you want it to, I think?

Reviewer 2: Thank you, I removed the link
